routes/booking_test.go

```diff
@@ -0,0 +1,422 @@
+}
+
+func (suite *AppTestSuite) Test_CreateBooking() {
+	// Check ID does not exist in database
+	t := suite.T()
+	rr := executeReq(t, &testRouteConfig{
+		Method:  http.MethodGet,
+		Body:    nil,
+		Handler: suite.app.GetAllBookings,
+		URL:     "/bookings/",
+	})
+	var payload []*model.Booking
+	_ = json.Unmarshal(rr.Body.Bytes(), &payload)
+	for _, b := range payload {
+		if BookingEqualMinusID(newBooking, b) {
+			t.Fatalf("testCreateBooking: %s found in booking list, when it should not exist", newBooking.ID)
+		}
+	}
+	// [POST] Create booking
+	requestBody, _ := json.Marshal(map[string]interface{}{
+		"workspace_id": newBooking.WorkspaceID,
+		"user_id":      newBooking.UserID,
+		"start_time":   newBooking.StartDate,
+		"end_time":     newBooking.EndDate,
+		"cancelled":    newBooking.Cancelled,
+		"created_by":   newBooking.CreatedBy,
+	})
+	rr2 := executeReq(t, &testRouteConfig{
+		Method:  http.MethodPost,
+		Body:    bytes.NewBuffer(requestBody),
+		Handler: suite.app.CreateBooking,
+		URL:     fmt.Sprintf("/bookings"),
+	})
+	// Check correct response
+
+	assert.Equal(t, rr2.Code, http.StatusCreated, "status code")
+	var payload2 *model.Booking
+	_ = json.Unmarshal(rr2.Body.Bytes(), &payload2)
+	// Just give it the ID since we want to use assert
+	newBooking.ID = payload2.ID
+	assert.Equal(t, newBooking, payload2, "The created booking is not the same as the sent request.")
+}
+
+var startBookingPatch, _ = utils.TimeStampToTime("7952342400")
+var endBookingPatch, _ = utils.TimeStampToTime("7952515199")
+var patchBooking = &model.Booking{
+	ID:          newBooking.ID, // Unknown at this point
+	UserID:      "8b5bb736-6a1d-3378-8e71-ab45fe8beb84",
+	WorkspaceID: "aad40cbb-4baf-3931-a5d2-6f98b414182a",
+	Cancelled:   true,
+	StartDate:   startBookingPatch,
+	EndDate:     endBookingPatch,
+	CreatedBy:   "e99a988a-1d41-3997-8d59-959a48ac24a0",
+}
+
+func (suite *AppTestSuite) Test_PatchBooking() {
+	t := suite.T()
+	// Assume ID exists since we just create it
+	// Change everything but the ID
+	requestBody, _ := json.Marshal(map[string]interface{}{
+		"workspace_id": patchBooking.WorkspaceID,
+		"user_id":      patchBooking.UserID,
+		"start_time":   patchBooking.StartDate,
+		"end_time":     patchBooking.EndDate,
+		"cancelled":    patchBooking.Cancelled,
+		"created_by":   patchBooking.CreatedBy,
+	})
+	rr2 := executeReq(t, &testRouteConfig{
+		Method:  http.MethodPatch,
+		Body:    bytes.NewBuffer(requestBody),
+		Handler: suite.app.UpdateBooking,
+		URL:     fmt.Sprintf("/bookings/%s", newBooking.ID),
+		URLParams: map[string]string{
+			"id": newBooking.ID,
+		},
+	})
+	// Response 200
+	var payload2 *model.Booking
+	_ = json.Unmarshal(rr2.Body.Bytes(), &payload2)
+	// Just give it the ID since we want to use assert
+	patchBooking.ID = payload2.ID
+	assert.Equal(t, patchBooking, payload2, "The created booking is not the same as the sent request.")
+}
+
+func (suite *AppTestSuite) Test_ZDeleteBooking() {
+	t := suite.T()
```

Comment:
would it be possible/helpful to collapse the Create/update and delete functionality into 1 test? Since its the only one with side-effects.. thoughts??

Reply:
Hmm. It's true that if one of these fails the others are going to fail anyhow..